pkg/netpol/eval/internal/k8s/netpol.go

```diff
@@ -407,3 +407,64 @@ func (np *NetworkPolicy) Selects(p *Pod, direction netv1.PolicyType) (bool, erro
 func (np *NetworkPolicy) fullName() string {
 	return types.NamespacedName{Name: np.Name, Namespace: np.Namespace}.String()
 }
+
+// ScanRulesForIngressFromEntireCluster scans ingress spec of the policy to check if the rules exposes given pod
+// to entire cluster; if yes updates the pod's connection to entire cluster field
+func (np *NetworkPolicy) ScanRulesForIngressFromEntireCluster(dst Peer) error {
+	// since we get here after scanning rules for allowed conns between peers, we can assume the rules are ok
+	for _, rule := range np.Spec.Ingress {
+		rulePeers := rule.From
+		rulePorts := rule.Ports
+		if !np.ruleSelectsEntireCluster(rulePeers) {
+			continue
+		}
+		ruleConns, err := np.ruleConnections(rulePorts, dst)
+		if err != nil {
+			return err
+		}
+		dst.GetPeerPod().updatePodXgressExposureToEntireClusterData(ruleConns, true)
+	}
+	return nil
+}
+
+// ScanRulesForEgressToEntireCluster scans egress spec of the policy to check if the rules exposes src peer
+// to entire class, if yes update egress to entire class connection details of the src
+// dst is provided for computing the correct connection of the rules
+func (np *NetworkPolicy) ScanRulesForEgressToEntireCluster(src, dst Peer) error {
+	for _, rule := range np.Spec.Egress {
+		rulePeers := rule.To
+		rulePorts := rule.Ports
+		if !np.ruleSelectsEntireCluster(rulePeers) {
+			continue
+		}
+		ruleConns, err := np.ruleConnections(rulePorts, dst)
+		if err != nil {
+			return err
+		}
+		src.GetPeerPod().updatePodXgressExposureToEntireClusterData(ruleConns, false)
+	}
+	return nil
+}
+
+// ruleSelectsEntireCluster checks if the given rules list is exposed to entire cluster;
+// i.e. if the rules list is empty or if there is a rule with empty namespaceSelector
+// since this func is definitely called after ruleSelectsPeer, there is no rules' correctness check here
+func (np *NetworkPolicy) ruleSelectsEntireCluster(rules []netv1.NetworkPolicyPeer) bool {
```

Comment:
it is possible to store per policy following data (for both ingress and egress )
```
allowsConnsToEntireWorld bool
connToEntireCluster *common.connectionSet
```
when a policy is added to the engine.

however I think this should be done in another PR (before optimizing generating representative pods)
because by scanning policy rules on while "building" the policy-engine for our required data, we should check the rules' correctness (legality)
this correctness check currently is done in `ruleSelectsPeer` (that's why called this after `ruleSelectsPeer` ) - but then correctness check will be eliminated from `ruleSelectsPeer`

Reply:
I see;
1. yes we can assume rules are valid in preprocessing;
another approach, if upserting policies for policy engine is the same also for `eval` command; we can always store the data for the policy (not only for exposure-analysis; and then will benefit from these data before scanning the policies) 
2. yes; if we'll store this data; while looping policies selecting a pod; we can update the pod's entireClusterConnection from this policy data (`ScanRulesForIngressFromEntireCluster` is deprecated in this case)

Reply:
1. it depends if when adding policies we activate the pre-processing only for exposure analysis..
2. right